zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/batchoperations/BatchOperationUpdateTask.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.tasks.batchoperations;
+
+import io.camunda.exporter.tasks.BackgroundTask;
+import io.camunda.exporter.tasks.batchoperations.BatchOperationUpdateRepository.DocumentUpdate;
+import io.camunda.webapps.schema.descriptors.operate.template.BatchOperationTemplate;
+import io.camunda.webapps.schema.entities.operation.BatchOperationEntity;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+
+public class BatchOperationUpdateTask implements BackgroundTask {
+
+  private final BatchOperationUpdateRepository batchOperationUpdateRepository;
+
+  private final Logger logger;
+
+  public BatchOperationUpdateTask(
+      final BatchOperationUpdateRepository batchOperationUpdateRepository, final Logger logger) {
+    this.batchOperationUpdateRepository = batchOperationUpdateRepository;
+    this.logger = logger;
+  }
+
+  @Override
+  public CompletionStage<Integer> execute() {
```

Comment:
❓  Do we have worries about any part of `execute` failing? I assume the `BatchOperationUpdateRepository` makes calls to something and is not entirely in memory, or will everything be caught and handled by the individual method themselves?

Reply:
That's a valid point! I just added another TODO item to the issue and I would like to take care of this after I make thigs asynchronous. The TODO item sounds like: "think through error processing. If some internal call (e.g. to ES/OS) fails, we must make sure that error is properly logged and task is retried"